Add search type link

Sola0404 opened the pull request:
<!-- Hey there. Thank you so much for improving NewPipe, and filling out the details. Having roughly the same layout helps everyone considerably :)-->

#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
<!-- While bullet points are the norm in this section, feel free to write free-form text instead of a list -->
- Add SEARCH type case in the intent handler
- Add a method to generate an intent conveying the information of service id, URL, link type, and search string. This method is used in getIntentByLink when link type is SEARCH, distinguishing the conditions between using SEARCH link and open search.

#### Fixes the following issue(s)
Fixes #2313 

#### PR relies on
https://github.com/TeamNewPipe/NewPipeExtractor/pull/950

#### APK testing
<!-- Use a new, meaningfully named branch. The name is used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe, e.g. "commentfix", if your PR implements a bugfix for comments. (No names like "patch-0" and "feature-1".)  -->
<!-- Remove the following line if you directly link the APK created by the CI pipeline. Directly linking is preferred if you need to let users test.-->
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

AudricV commented:
Could you explain better what you added in this pull request? Also, it seems you did extractor changes for this, but you didn't use them in this PR. Please do so if that's needed.

Sola0404 commented:
Thank you for your response. I've updated the description. 
I changed extractor for this issue, but I don't know how to include these changes in this PR, cause they are from different repositories. Could you give me some advice? Thanks in advance!

Stypox commented:
You need to create a pull request in NewPipeExtractor, too. As you noticed, they are two different repositories, hence two different PRs ;-)

ShareASmile commented:
> @Sola0404 You need to create a pull request in NewPipeExtractor, too.

@Stypox it's already been opened on extractor https://github.com/TeamNewPipe/NewPipeExtractor/pull/950

AudricV wrote:
> it seems you did extractor changes for this, but you didn't use them in this PR. Please do so if that's needed.

Stypox @Sola0404 is asking how to use extractor commits in this PR

Stypox commented on app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java:
```diff
@@ -672,6 +686,11 @@ public static Intent getIntentByLink(final Context context,
         }
 
+        if (linkType == StreamingService.LinkType.SEARCH) {
+            return getOpenSearchIntent(context, url, service.getServiceId(),
+                    linkType, service.getIdByUrl(url));
```
Intead of obtaining the ID this way, as said in https://github.com/TeamNewPipe/NewPipeExtractor/pull/950#discussion_r1006825163, I think you should use `service.getSearchLHFactory().getId(url)`.

Stypox reviewed (changes requested):
Are urls copied to the search field in NewPipe handled as opened URLs?

Stypox commented:
I'm closing this for now, both because of inactivity, and because we're going to start the rewrite soon and this would get conflicts despite its simplicity. I'm leaving https://github.com/TeamNewPipe/NewPipeExtractor/pull/950#issuecomment-2025494747 open though, so once that is done and the rewrite is in a good shape this can be reopened. Thank you!
